typegraph/node/sdk/src/typegraph.ts

```diff
@@ -212,25 +212,20 @@ export function typegraph(
 
   builder(g);
 
-  let serialize = () => "";
-  if (!disableAutoSerialization) {
-    const tgJson = core.finalizeTypegraph({ tag: "simple" });
-    console.log(tgJson);
-    serialize = () => tgJson;
-  } else {
-    // config is known at deploy time
-    serialize = (config?: ArtifactResolutionConfig) => {
-      const tgJson = core.finalizeTypegraph({
-        tag: "resolve-artifacts",
-        val: config,
-      });
+  const ret = {
+    serialize(config: ArtifactResolutionConfig) {
+      const tgJson = core.finalizeTypegraph(config);
       return tgJson;
-    };
-  }
-  return {
-    serialize,
+    },
     name,
-  };
+  } as TypegraphOutput;
+
+  if (Manager.isRunFromCLI()) {
```

Comment:
thought: maybe supply a second function for the auto-deploying method or are there cases where one might use the same `typegraph` through tgDeploy and CLI?

Reply:
What I meant was to provide two versions of the `typegraph` function itself. The default one that auto-deploys and expects to be invoked from the CLI and another that returns the typegraph desc for `tgDeploy` to consume. The assumption here being that a user who's deploying through the CLI would probably not be able to use tgDeploy on the same `typegraph` fn invocation.

Upon further thought though, maybe test only users `tgDeploy` might want both functionality.